pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +375,530 @@ def Sphere(
+        points. Must have at least two points and be non-negative.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying ``theta``
+        points. Must have at least two points.  Can have any value as long
+        as range is within 360 degrees. Large magnitudes may result in
+        problems with endpoint overlap detection.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying ``phi``
+        points. Must have at least two points.  Must be between
+        0 and 180 degrees.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center coordinate vector in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction coordinate vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees ``phi``.
+
+    radians : bool, default: False
+        Whether to use radians for ``theta`` and ``phi``.
+
+    tolerance : bool, default: 1.0e-8
+        Absolute tolerance for endpoint detection for ``theta``, ``phi`,
+        and ``radius``.
+
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.SolidSphere: Sphere creation using linear sampling.
+    pyvista.Sphere: Sphere that describes outer 2D surface.
+
+    Examples
+    --------
+    Create a solid sphere using default parameters.  The
+    default creates a sphere that is linearly sampled with
+    respect to spherical coordinates, the same as
+    :func:`pyvista.SolidSphere`.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> solid_sphere = pv.SolidSphereGeneric()
+    >>> solid_sphere.plot(show_edges=True)
+
+    Linearly sampling spherical coordinates does not lead to
+    cells of all the same size at each radial position.  In
+    particular, cells near the poles have smaller sizes.
+    Plot the cell volumes of a solid sphere with only one layer of
+    cells.
+
+    >>> solid_sphere = pv.SolidSphereGeneric(
+    ...     radius=np.linspace(0, 0.5, 2),
+    ...     theta=np.linspace(180, 360, 30),
+    ...     phi=np.linspace(0, 180, 30),
+    ... )
+    >>> solid_sphere = solid_sphere.compute_cell_sizes()
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    Sampling the polar angle in a nonlinear manner allows for consistent cell volumes.  See
+    `Sphere Point Picking <https://mathworld.wolfram.com/SpherePointPicking.html>`_.
+
+    >>> phi = np.rad2deg(np.arccos(np.linspace(1, -1, 30)))
+    >>> solid_sphere = pv.SolidSphereGeneric(
+    ...     radius=np.linspace(0, 0.5, 2),
+    ...     theta=np.linspace(180, 360, 30),
+    ...     phi=phi,
+    ... )
+    >>> solid_sphere = solid_sphere.compute_cell_sizes()
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    """
+    if radius is None:
```

Comment:
Set default tolerance to be None, and then set tolerance here based on if radians is used.  Scale tolerance if user supplies and radians are not used.

Reply:
I decided that splitting the tolerance behavior to be in different units makes things too complex.  It also means that you might get different results defining something in degrees vs. radians without also changing the tolerance.  Since this feature is likely so rarely to be used, I think it is best just to document which units are used for comparison.

Reply:
Weak preference here since I agree this feature is in the weeds:

I can see you've gone back and forth on this. I guess there are a few things being weighed:

1. It's annoying to have `tolerance=None` in the signature because then the function body and documentation get a bit more complicated.
2. It would be nice if the user changing their input from degrees <-> radians and changing the `radians` flag accordingly ALWAYS results in the same results (i.e. a complete sphere or not).
3. It would be nice if the `tolerance` input matched the users input units. So if the user inputs angles in degrees but the tolerance is specified in radians that's a bit confusing.

The solution you have right now (`1e-08` in the signature, but always interpreted as radians) optimizes for points 1 and 2 above but fails for point 3. My preference would be to optimize for points 2 and 3 and accept the complexity from point 1. 

```
if tolerance is None:
    tolerance = 1e-08
    if radians:
        tolerance = np.deg2rad(tolerance)        
```

It sounds like you considered this solution but decided to reject it. No problem if your conclusion still stands.

Also, I guess ears should be open for a user that wants to control the 4 different tolerances independently. If someone wants that for some reason they do NOT have that control in the current code. But it's hard to see use cases where users want spheres that are < 1e-07 away from touching the origin, polar axis, or prime meridian. The use case I see is that users are having a hard time handling numerical issues occurring in their data at these boundaries and they're just trying exclude them for ease. But even then, I'd expect gaps in theta and phi at the 1e-02 or 1e-03 level unless they sampling like millions of points in theta and phi, which again, I think is unlikely. I think it's mostly likely to see the tolerance hit if users are trying to make tiny spheres (like micron scale) without intelligently scaling up units first.

Reply:
```py
if tolerance is None:
    tolerance = 1e-08
    if radians:
        tolerance = np.deg2rad(tolerance)        
```

This definition doesn't make sense to me.  This says that the tolerance is 1e-8 when `radians is False`.  When `radians is True`, tolerance is 1.7e-10.  Note that this number is compared to a value in radians, so the end value is in radians.  So it results in neither tolerance matching the users expected units and doesn't satisfy point 2.

There is also the below, resulting in 1e-8 tolerance for radians and 1e-8 tolerance in degrees converted to radians.  But this also doesn't satisfy point 2.  The only way to satisfy point 2 is to have different default starting tolerance depending on whether it is radians or degrees that is converted to the same tolerance in radians.  We'd have to document something like `the default tolerance is 1e-8 in radians and 1e-8 / 180 * np.pi in degrees, which results in a tolerance of 1e-8 radians`.  This seems like a more complicated way to end up in the same position to me but willing to hear more opinions on it.

```py
if tolerance is None:
    tolerance = 1e-08
    if not radians:
        tolerance = np.deg2rad(tolerance)
```


A bigger issue in my opinion is that we have the same tolerance for angles and for radius.  Let's give two options, which seems like it will now cover 99.9999% of cases. For those .0001% cases, they can copy the source code and adjust it :)

Reply:
Yeah, it's an annoying documentation nightmare. The issue with the tolerance being passed in always meaning radians is: suppose a user really wants a sphere to end 2e-08 *degrees* away from one of the axes. they now have to wonder "is 2e-08 degrees bigger or smaller than 1e-08 radians?" So this is pretty annoying for them. I guess they can use `np.deg2rad` on whatever tolerance they want to select just to be careful.

Consider the phrasing for the docstring:

```
Absolute tolerance (specified in radians) for endpoint detection for ``phi`` and ``theta``.
```